src/30_Advanced/Copy_Button.html

```diff
@@ -0,0 +1,93 @@
+while a placeholder button remains visible.
+-->
+<!-- -->
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+
+  <!-- ## Setup -->
+  <!-- We use `amp-iframe` to load a copy button, replacing a placeholder button.-->
+  <script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>
+
+  <link rel="canonical" href="<%host%>/advanced/copy_button/">
+  <title>Copy Button</title>
+
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+
+  <style amp-custom>
+/* common styles for button */
+button#copy{
+  box-sizing: border-box;
+  font: inherit;
+  font-size: 18px;
+  line-height: 2em;
+  cursor: pointer;
+  border-radius: 4px;
+  border: 2px solid transparent;
+  background: #555;
+  padding: 0 8px;
+  color: #fff;
+  display: block;
+}
+/* disable for placeholder */
+button#copy {
+  cursor: not-allowed;
+  width: 100%;
+  height: 100%;
+}
+/* misc page CSS */
+body {
+  font-family: Sans-Serif;
+}
+main {
+  display: flex;
+  align-items: center;
+  padding: 1em;
+}
+main span {
+  margin: 0 1em;
+}
+  </style>
+</head>
+<body>
+
+<main>
+  <span>
+    Copy Me!
+  </span>
+
+  <!-- ## How a "Copy" button works -->
+  <!-- By including `amp-iframe` with custom JS (this is in `Copy_Button__copier.html`), we can
+       perform any operation from vanilla JS, including copy.
+       The iframe is allowed to load anywhere on your AMP as it has a placeholder, and in our case,
+       this placeholder has exactly the same presentation as the final display so it's not obvious
+       when it is "swapped out". Users who hover over the button before it loads will see a
+       "disabled" indicator.
+       We pass the data to be copied via the `#` param. The copier "helper" page can always serve
+       the same HTML and can be cached in its own right.
+  -->
+  <amp-iframe
+      sandbox="allow-scripts"
+      width="64"
+      height="42"
+      frameborder="0"
+      src="<% api.host %>/iframe/resizable-iframe.html#This text was copied from &quot;AMP by Example&quot;!">
```

Comment:
s/resizable-iframe/copier

Reply:
One other thing: amp-iframes need to be served from a different host if the `allow-same-origin` sandbox attribute is used. This is why we have a different `api.host`. In this case that is not necessary and I think it's better to serve the iframe from the same domain (to not indicate that this might be required). You can simply copy `copier.html` into the `/static` dir and then reference it via `/copier.html'.

Reply:
Done